pkg/operator/controller/garden/garden/components.go

```diff
@@ -618,10 +619,13 @@ func (r *Reconciler) newKubeAPIServer(
 				AuthorizedTTL:                            metav1.Duration{Duration: time.Duration(0)},
 				UnauthorizedTTL:                          metav1.Duration{Duration: time.Duration(0)},
 				Timeout:                                  metav1.Duration{Duration: 10 * time.Second},
-				FailurePolicy:                            apiserverv1beta1.FailurePolicyNoOpinion,
+				FailurePolicy:                            apiserverv1beta1.FailurePolicyDeny,
 				SubjectAccessReviewVersion:               "v1",
 				MatchConditionSubjectAccessReviewVersion: "v1",
-				MatchConditions:                          nil,
+				MatchConditions: []apiserverv1beta1.WebhookMatchCondition{{
+					// only intercept request from gardenlets and service accounts from seed namespaces
+					Expression: fmt.Sprintf("'%s' in request.groups || request.groups.exists(e, e.startsWith('%s%s'))", v1beta1constants.SeedsGroup, serviceaccount.ServiceAccountGroupPrefix, gardenerutils.SeedNamespaceNamePrefix),
+				}},
```

Comment:
With this change we could try to get rid of deploying the virtual-kubeapiserver twice. There should be no calls to these namespaces until the Garden was reconciled for the first time.
I don't know if it works and it could be verified and implemented in an own PR.

Reply:
That's a good point. I added a TODO statement. I think we can only do it once we only support Kubernetes versions of 1.32+.